org.adempiere.base.process/src/org/compiere/process/CreateFromInOut.java

```diff
@@ -113,52 +114,25 @@ private String createLines() {
 					selectionIDList.add(T_Selection_ID);
 				
 				String ColumnName = "AD_Table_ID";
-				String key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "Line";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "C_Order_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "C_Invoice_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "M_RMA_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "Qty";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getBigDecimal(ColumnName));
-				
-				ColumnName = "C_UOM_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "M_Locator_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "M_Product_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "C_OrderLine_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "C_InvoiceLine_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
-				
-				ColumnName = "M_RMALine_ID";
-				key = ColumnName + "_" + T_Selection_ID;
-				selectionValueMap.put(key, rs.getInt(ColumnName));
+				MTable table = MTable.get(rs.getInt(ColumnName));
+				if (table.getAD_Table_ID() == MOrderLine.Table_ID)
+				{
+					ColumnName = "C_OrderLine_ID";
+					String key = ColumnName + "_" + T_Selection_ID;
+					selectionValueMap.put(key, T_Selection_ID);
+				}
+				else if (table.getAD_Table_ID() == MInvoiceLine.Table_ID)
+				{
+					ColumnName = "C_InvoiceLine_ID";
+					String key = ColumnName + "_" + T_Selection_ID;
+					selectionValueMap.put(key, T_Selection_ID);
+				}
+				else if (table.getAD_Table_ID() == MRMALine.Table_ID)
+				{
+					ColumnName = "M_RMALine_ID";
+					String key = ColumnName + "_" + T_Selection_ID;
+					selectionValueMap.put(key, T_Selection_ID);
+				}
```

Comment:
this can be simplify to
```
MTable table = MTable.get(rs.getInt(ColumnName));
String key = table.getKeyColumns()[0] + "_" + T_Selection_ID;
selectionValueMap.put(key, T_Selection_ID);
```